prelude/cxx/windows_resource.bzl

```diff
@@ -0,0 +1,62 @@
+load("@prelude//cxx:cxx_context.bzl", "get_cxx_toolchain_info")
+load("@prelude//linking:link_info.bzl", "LinkInfo", "LinkInfos", "create_merged_link_info", "LibOutputStyle", "ObjectsLinkable")
+load("@prelude//linking:linkable_graph.bzl", "create_linkable_graph")
+load("@prelude//linking:link_groups.bzl", "LinkGroupLibInfo")
+load("@prelude//linking:shared_libraries.bzl", "SharedLibraryInfo")
+
+
+def windows_resource_impl(ctx: AnalysisContext) -> list[Provider]:
+    objects = []
+
+    toolchain = get_cxx_toolchain_info(ctx)
+    for src in ctx.attrs.srcs:
+        rc_output = ctx.actions.declare_output(
+            "__objects__",
+            "{}.res".format(src.short_path),
+        )
+        rc_cmd = cmd_args(toolchain.rc_compiler_info.compiler)
+        rc_cmd.add(cmd_args(rc_output.as_output(), format = "/fo{}"))
+        rc_cmd.add(src)
+
+        ctx.actions.run(
+            rc_cmd,
+            category = "rc_compile",
+        )
+
+        cvtres_output = ctx.actions.declare_output(
+            "__objects__",
+            "{}.obj".format(src.short_path),
+        )
+        cvtres_cmd = cmd_args(toolchain.cvtres_compiler_info.compiler)
+        cvtres_cmd.add(cmd_args(cvtres_output.as_output(), format = "/OUT:{}"))
+        cvtres_cmd.add(rc_output)
+
+        ctx.actions.run(
+            cvtres_cmd,
+            category = "cvtres_compile",
+        )
+
+        objects.append(cvtres_output)
+
+    link = LinkInfo(
+        name = ctx.attrs.name,
+        linkables = [ObjectsLinkable(
+            objects = objects,
+            linker_type = toolchain.linker_info.type,
+            link_whole = True,
+        )],
+    )
+
+    providers = [
+        DefaultInfo(default_output = None),
+        SharedLibraryInfo(set = None),
+        LinkGroupLibInfo(libs = {}),
```

Comment:
Are these empty providers really needed?

Reply:
Yes, at least from what I know.
If I remove the first one the error is in:
```
prelude/linking/linkables.bzl:42:31
    shared_library_info = dep[SharedLibraryInfo],
```
And if I remove the second one the error is in:
```
prelude/linking/linkables.bzl:40, in linkable
    expect(LinkGroupLibInfo in dep, str(dep.label.raw_target()))
```